### DISCUSSION TITLE
postgrest-js: Need a way to identify one-to-one relation for better return-type deduction

### DISCUSSION THREAD

**Discussion by tvogel** [supabase/supabase#12256] [Feature Requests]
Currently, the generated database typescript types do not include any information about unique foreign-key relations. Thus, when doing a `.select('foreign_table(column)')`, the returned `data` will be of type 
```JS
{ 
  foreign_table: 
    { column: column_type } 
  | { column: column_type }[] 
  | null 
}
```
So, I need to manually exclude the array and null case from prior knowledge.

Is there any plan to either include foreign-key relations in the generated database types or at least allow manual annotation in the `.select()` query-syntax to allow to indicate, there will be exactly one result for the queried foreign column?

**Comment by barbinbrad** [comment]
Bumping this. Same question. There’s a lot of boilerplate (or @ts-ignores) required to do a simple join. Hope I’m just doing something wrong
